File: 692. Top K Frequent Words/topk.py
class Solution:
    def topKFrequent(self, words: List[str], k: int) -> List[str]:
        word_count = {}
        for word in words:
            if word in word_count:
                word_count[word] += 1
            else:
                word_count[word] = 1
        word_count = sorted(word_count.items(), key=lambda x: (-x[1], x[0]))
        return [word_count[i][0] for i in range(k)]

File: 692. Top K Frequent Words/topk.cpp
class Solution {
public:
    vector<string> topKFrequent(vector<string>& words, int k) {
        unordered_map<string, int> word_count;
        for (auto word: words)
            word_count[word]++;
        vector<pair<string, int>> word_count_vec(word_count.begin(), word_count.end());
        sort(word_count_vec.begin(), word_count_vec.end(), 
            [](const pair<string, int>& a, const pair<string, int>& b) {
            return a.second == b.second ? a.first < b.first : a.second > b.second;
        });
        vector<string> res;
        for (int i = 0; i < k; i++)
            res.push_back(word_count_vec[i].first);
        return res;
    }
};